app/views/questionnaire.py

```diff
@@ -137,27 +134,23 @@ def post_block(routing_path, schema, metadata, collection_metadata, answer_store
 
     block = schema.get_block(current_location.block_id)
 
-    placeholder_renderer = PlaceholderRenderer(block, answer_store=answer_store, metadata=metadata)
-    replaced_block = placeholder_renderer.render()
+    transformed_block = transform_variants(block, schema, metadata, answer_store)
 
-    transformed_block = transform_variants(replaced_block, schema, metadata, answer_store)
+    placeholder_renderer = PlaceholderRenderer(answer_store=answer_store, metadata=metadata)
+    replaced_block = placeholder_renderer.render(transformed_block)
```

Comment:
I understand that we don't want to confuse this with the old way of rendering a block, but given that we're calling a render method with a block, `rendered_block` would be clearer than `replaced_block`.